machida/lib/wallaroo/experimental/__init__.py

```diff
@@ -93,10 +93,10 @@ def connect(self, host=None, port=None):
                 else:
                     raise
 
-    def write(self, message, event_time=0):
+    def write(self, message, key=None, event_time=0):
         if self._conn == None:
             raise RuntimeError("Please call connect before writing")
-        payload = self._encoder.encode(message, event_time)
+        payload = self._encoder.encode(message, key, event_time)
```

Comment:
ditto regarding the order here.
For internal methods, I would prefer to use named variables where possible to reudce the risk or assigning variables to the wrong parameters:
e.g.
`.encode(message, event_time=event_time, key=key)`